kernel.c

```diff
@@ -87,11 +179,10 @@ void idt_init(void)
 	write_port(0xA1 , 0xff);
 
 	/* fill the IDT descriptor */
-	idt_address = (unsigned long)IDT ;
-	idt_ptr[0] = (sizeof (struct IDT_entry) * IDT_SIZE) + ((idt_address & 0xffff) << 16);
-	idt_ptr[1] = idt_address >> 16 ;
+	idt_ptr.limit = sizeof(IDT);
```

Comment:
Shouldn't this be `sizeof(IDT) - 1` ?

Reply:
Yes I think you're right, but FWIW it shouldn't matter _that_ much since nothing should read past the end of the IDT or GDT (and if they did, are likely to read more than one byte).